lib/auth/auth.go

```diff
@@ -1685,7 +1642,7 @@ func (a *Server) GenerateServerKeys(req GenerateServerKeysRequest) (*PackedKeys,
 		}
 	}
 
-	isAdminRole := req.Roles.Equals(types.SystemRoles{types.RoleAdmin})
+	isAdminRole := reqRoles.Equals(types.SystemRoles{types.RoleAdmin})
```

Comment:
Do a contains check instead of an equals?

Reply:
When it comes to deciding whether or not to grant admin privileges I'm inclined to not change the existing behavior in any way without strong reason to.

Reply:
Fair enough.